event_corrector/main_outline_zarr.py

```diff
@@ -691,7 +721,78 @@ def _connect_widget(self):
+        Create a back up of the labels if not existing
+        """
+        if f"{self.folder_type}_backup" not in self.path_D2:
+            print("No back-up found, creating one ...")
+            self.animal.IMAGE.D2.create_dataset(
+                name=f"{self.folder_type}_backup",
+                shape=self.labels.shape,
+                dtype="uint8",
+                chunks=(1, *self.labels.shape[1:]),
+            )
+        backup_label = self.path_D2[f"{self.folder_type}_backup"]
+        for t in tqdm(range(self.labels.shape[0]), desc="Checking/generating back up"):
+            if np.any(backup_label[t]):
+                continue
+            backup_label[t] = self.labels[t].copy()
+
+    def ensure_skeleton_exists(self):
+        """
+        Get the outlines saved or create them if not existing
+        """
+        if f"{self.folder_type}_skeleton" not in self.path_D2:
+            print(f"{self.folder_type} skeleton not found in Zarr — generating outlines...")
+            self.path_D2.create_dataset(
+                name=f"{self.folder_type}_skeleton",
+                shape=self.labels.shape,
+                dtype="uint8",
+                chunks=(1, *self.labels.shape[1:]),
+            )
+        self.skeleton_store = self.path_D2[f"{self.folder_type}_skeleton"]
+        for t in tqdm(range(self.labels.shape[0]), desc="Checking/generating skeleton"):
+            if np.any(self.skeleton_store[t]):
+                continue
+            skeleton = masks_to_outlines(self.labels[t]).astype(np.uint8)
+            self.skeleton_store[t] = skeleton
+
+    def change_folder_mode(self):
+        """
+        Change the current folder of labels
+        """
+        # Change folder_type based on the current selection and clear existing layers
+        self.folder_type = self.ui_widget.choose_folder_type.currentText()
+        print(f"current zarr: {self.folder_type}")
+        
+        # Remove any existing layers related to the selected folder_type and outlines
+        if self.folder_type in self.viewer.layers:
+            self.viewer.layers.remove(self.viewer.layers[self.folder_type])
+        outlines_name = f"outlines_{self.folder_type}"
+        if outlines_name in self.viewer.layers:
+            self.viewer.layers.remove(self.viewer.layers[outlines_name])
+
+        # Load the labels associated with the selected folder_type
+        self.labels_original = self.animal.IMAGE.D2[self.folder_type]
+        self.labels = self.animal.IMAGE.D2[self.folder_type][:].copy()
+        self.labels_layer = self.viewer.add_labels(self.labels, name=self.folder_type)
+
+        # Ensure the backup exists before proceeding
+        self.ensure_backup_exists()
+
+        # Check if skeleton exists and load it if available
+        self.ensure_skeleton_exists()
+        skeleton_zarr = self.skeleton_store[:].copy()  # Retrieve the skeleton data
+        self.outlines_layer = self.viewer.add_labels(skeleton_zarr, name=outlines_name)
```

Comment:
Je pense tu peux mettre ça dans une fonction et la reutiliser dans l'init 
```suggestion
        if self.folder_type in self.viewer.layers:
            self.viewer.layers.remove(self.viewer.layers[self.folder_type])
        outlines_name = f"outlines_{self.folder_type}"
        if outlines_name in self.viewer.layers:
            self.viewer.layers.remove(self.viewer.layers[outlines_name])

        # Load the labels associated with the selected folder_type
        self.labels_original = self.animal.IMAGE.D2[self.folder_type]
        self.labels = self.animal.IMAGE.D2[self.folder_type][:].copy()
        self.labels_layer = self.viewer.add_labels(self.labels, name=self.folder_type)

        # Ensure the backup exists before proceeding
        self.ensure_backup_exists()

        # Check if skeleton exists and load it if available
        self.ensure_skeleton_exists()
        skeleton_zarr = self.skeleton_store[:].copy()  # Retrieve the skeleton data
        self.outlines_layer = self.viewer.add_labels(skeleton_zarr, name=outlines_name)
```